KK-DZ/src/app/modules/workers/services/workers.service.ts

```diff
@@ -0,0 +1,72 @@
+import { Injectable } from '@angular/core';
+
+import { EmployeeEntity } from '../../../core/models/employee-entity';
+import { EmployeeFiltersState } from '../../../core/models/employee-filters-state';
+
+import { BehaviorSubject, combineLatest, Observable } from 'rxjs';
+import { map } from 'rxjs/operators';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class WorkersService {
+  private workers$: BehaviorSubject<EmployeeEntity[]>;
+  private filtersSubject$: BehaviorSubject<EmployeeFiltersState> = new BehaviorSubject({});
+
+  public list$: Observable<EmployeeEntity[]>;
+  public filters$ = this.filtersSubject$.asObservable()
+    .pipe(map(filters => Object.keys(filters).reduce((acc, key) => ({
+      ...acc,
+      ...(!!filters[key] ? {[key]: filters[key]} : {})
+    }), {})));
+
+  constructor() {
+    this.workers$ = new BehaviorSubject([]);
+
+    this.list$ = combineLatest([this.workers$, this.filters$]).pipe(
+      map(([workers, filters]) =>
+        workers.filter((worker: EmployeeEntity) => this.checkWorkerIsValid(worker, filters)
+        )
+      )
+    );
+  }
+
+  private checkWorkerIsValid(
+    worker: EmployeeEntity,
+    filters: EmployeeFiltersState
+  ): boolean {
+    return Object.keys(filters).every(
+      (key: string) => !isNaN(filters[key]) ?
+        Number(filters[key]) === Number(worker[key]) : `${worker[key]}`.toLowerCase().startsWith(`${filters[key]}`.toLowerCase()) 
+    );
+  }
+
+  public initialize(workers: EmployeeEntity[]) {
+    this.workers$.next(workers);
+  }
+
+  public add(worker: EmployeeEntity): void {
+    this.workers$.next([...this.workers$.value, worker]);
+  }
+
+  public edit(workerId: number, worker: EmployeeEntity) {
+    let editedWorkers = this.workers$.value;
```

Comment:
do workers$.next przekazujesz tę samą tablicę, ta sama referencja, przez co Angular może w pewnych miejscach ją wykorzystujących żle zadziałać, co jest związane z detekcją zmianm o której jeszcze w tym tygodniu porozmawiamy
```suggestion
    let editedWorkers = [...this.workers$.value];
```

ponadto let jest tu nie potrzebny, użyj const'a